Lib/asyncio/base_events.py

```diff
@@ -424,7 +424,11 @@ def create_task(self, coro, *, name=None):
 
         Return a task object.
         """
-        self._check_closed()
+        try:
+            self._check_closed()
+        except RuntimeError:
+            coro.close()
+            raise
```

Comment:
This solution seems too general for the specific issue in the bug. I would have expected that you'd modify `_ensure_future()` to keep track of whether it's called `_wrap_awaitable()` and if so, close the created task if the following code fails. What you do here might close a Future that was not created by `_wrap_awaitable()` and whose owner should be responsible for closing it.

Reply:
Fixed